HeavyNeutralLeptonAnalysis/plugins/HeavyNeutralLeptonAnalysis.cc

```diff
@@ -98,84 +103,296 @@ using namespace std;
-      virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
-      virtual void initialize(const edm::Event&); 
-      virtual void endJob() override;
-
-
+public:
+  explicit HeavyNeutralLeptonAnalysis(const edm::ParameterSet&);
+  ~HeavyNeutralLeptonAnalysis();
+  
+  bool PrimaryVertex( const reco::VertexCollection &vtx);
+  bool isAncestor(const reco::Candidate* ancestor, const reco::Candidate* particle);
+  double MatchGenFirstMuon(const edm::Event&,  reco::TrackRef BestTrack);
+  double MatchGenSecondMuon(const edm::Event&,  reco::TrackRef BestTrack);
+  double MatchGenVertex(const edm::Event& iEvent, reco::Vertex vertex);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  
+  
+private:
+  virtual void beginJob() override;
+  virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+  virtual void initialize(const edm::Event&); 
+  virtual void endJob() override;
+  
+  
 
- edm::Service<TFileService> fs;
- 
+  edm::Service<TFileService> fs;
   TTree * tree_;
- BigNtuple ntuple_;
-
-
-//--------------Variables------------------------
-
+  BigNtuple ntuple_;
+
+  int    debug;
+  float ivfMatchingScore;
+  float pvCompatibilityScore = .05;
+  bool  selIVFIsPV;
+
+  //--------------Variables------------------------
+  //======================= Primary Vertex Information ========================//                     
+  std::vector<float> pvX;
+  std::vector<float> pvY;
+  std::vector<float> pvZ;
+  std::vector<float> pvXErr;
+  std::vector<float> pvYErr;
+  std::vector<float> pvZErr;
+  std::vector<float> pvMass;
+  std::vector<float> pvLxy;
+  std::vector<float> pvLxyz;
+  std::vector<float> pvLxySig;
+  std::vector<float> pvLxyzSig;
+  std::vector<float> pvChi2;
+  std::vector<int> pvNTrack;
+  std::vector<float> pvSumPtSq;
+  std::vector<float> NumberPV;
+
+  //============= Trigger Information ===========================//  
+  static const Int_t MAX_TRIGGERS      = 200;
+  Int_t  nTrig;
+  std::vector<std::string>   triggerNames;
+  Int_t         triggerPass[MAX_TRIGGERS];
+  Int_t           passIsoTk18 ;
+  Int_t           passIsoTk20 ;
+  Int_t           passIsoTk22 ;
+  Int_t           passIsoTk24 ;
+  Int_t           passIsoTk27 ;
+  Int_t           passIsoTk17e;
+  Int_t           passIsoTk22e;
+
+  Int_t           passIsoMu18 ;
+  Int_t           passIsoMu20 ;
+  Int_t           passIsoMu22 ;
+  Int_t           passIsoMu24 ;
+  Int_t           passIsoMu27 ;
+  Int_t           passIsoMu17e;
+  Int_t           passIsoMu22e;
+  Int_t           passTkMu17  ;
+  Int_t           passTkMu20  ;
+
+
+  Int_t           passIsoMu24All;
+  Int_t           passIsoMu27All;
+
+  Int_t           passDoubleMu17TrkIsoMu8    ;
+  Int_t           passDoubleMu17TrkIsoTkMu8  ;
+  Int_t           passDoubleTkMu17TrkIsoTkMu8;
+  
+  //============= Secondary Vertex Information ===========================//
+
+  std::vector<int>   vertex_TrackSize;
```

Comment:
@jpriscia @Moanwar @hayfasfar, branches are to be created in the `BigNtuple.h` file